core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java

```diff
@@ -316,39 +324,40 @@ protected PipeRunResult executeStatementSet(QueryExecutionContext queryExecution
-					String contentDisposition = (String) session.get("contentDisposition");
-					return executeSelectQuery(statement,blobSessionVar,clobSessionVar, response, contentType, contentDisposition, session, next);
-				} else {
-					return executeSelectQuery(statement,blobSessionVar,clobSessionVar, session, next);
-				}
-			} 
-			if ("updateBlob".equalsIgnoreCase(queryExecutionContext.getQueryType())) {
-				if (StringUtils.isNotEmpty(getBlobSessionKey())) {
-					return new PipeRunResult(null, executeUpdateBlobQuery(statement,session==null?null:Message.asMessage(session.get(getBlobSessionKey()))));
-				} 
-				return new PipeRunResult(null, executeUpdateBlobQuery(statement, message));
-			} 
-			if ("updateClob".equalsIgnoreCase(queryExecutionContext.getQueryType())) {
-				if (StringUtils.isNotEmpty(getClobSessionKey())) {
-					return new PipeRunResult(null, executeUpdateClobQuery(statement,session==null?null:Message.asMessage(session.get(getClobSessionKey()))));
-				} 
-				return new PipeRunResult(null, executeUpdateClobQuery(statement, message));
-			} 
-			if ("package".equalsIgnoreCase(queryExecutionContext.getQueryType())) {
-				return new PipeRunResult(null, executePackageQuery(queryExecutionContext));
+			switch(queryExecutionContext.getQueryTypeEnum()) {
+				case SELECT:
+					Object blobSessionVar=null;
+					Object clobSessionVar=null;
+					if (session!=null && StringUtils.isNotEmpty(getBlobSessionKey())) {
+						blobSessionVar=session.get(getBlobSessionKey());
+					}
+					if (session!=null && StringUtils.isNotEmpty(getClobSessionKey())) {
+						clobSessionVar=session.get(getClobSessionKey());
+					}
+					if (isStreamResultToServlet()) {
+						HttpServletResponse response = (HttpServletResponse) session.get(PipeLineSession.HTTP_RESPONSE_KEY);
+						String contentType = (String) session.get("contentType");
+						String contentDisposition = (String) session.get("contentDisposition");
+						return executeSelectQuery(statement,blobSessionVar,clobSessionVar, response, contentType, contentDisposition, session, next);
+					} else {
+						return executeSelectQuery(statement,blobSessionVar,clobSessionVar, session, next);
+					}
+				case UPDATEBLOB:
+					if (StringUtils.isNotEmpty(getBlobSessionKey())) {
+						return new PipeRunResult(null, executeUpdateBlobQuery(statement,session==null?null:Message.asMessage(session.get(getBlobSessionKey()))));
+					} 
+					return new PipeRunResult(null, executeUpdateBlobQuery(statement, message));
+				case UPDATECLOB:
+					if (StringUtils.isNotEmpty(getClobSessionKey())) {
+						return new PipeRunResult(null, executeUpdateClobQuery(statement,session==null?null:Message.asMessage(session.get(getClobSessionKey()))));
+					} 
+					return new PipeRunResult(null, executeUpdateClobQuery(statement, message));
+				case PACKAGE:
+					return new PipeRunResult(null, executePackageQuery(queryExecutionContext));
+				default:
+					break;
```

Comment:
It is a good practice to 'do' something here, like:
- comment or print a log line that this case is expected and does not need special handling
- throw an IllegalStateException, if this case should not be possible. That helps to fix problems when OutputTypes change over the years